js/src/forum/checkForSuspension.ts

```diff
@@ -0,0 +1,16 @@
+import app from 'flarum/forum/app';
+import SuspensionInfoModal from './components/SuspensionInfoModal';
+
+export default function () {
+  return new Promise(() => {
+    setTimeout(() => {
+      if (app.session.user) {
+        const message = app.session.user.suspendMessage();
+        const until = app.session.user.suspendedUntil();
+        if (message) {
+          app.modal.show(SuspensionInfoModal, { message, until });
+        }
+      }
+    }, 1000);
+  });
+}
```

Comment:
this will show on every single page load ? that makes the forum unusable for the suspended user, we should store a value in localStorage to determine if the modal has been shown once.